csharp/src/Drivers/Databricks/Reader/CloudFetch/DownloadResult.cs

```diff
@@ -103,10 +103,14 @@ public void UpdateWithRefreshedLink(TSparkArrowResultLink refreshedLink)
         }
 
         /// <inheritdoc />
-        public void SetCompleted(Stream dataStream, long size)
+        public void SetCompleted(ReadOnlyMemory<byte> data, long size)
```

Comment:
With this change, we probably don't need to take `size` separately anymore. That said, if any of these uncompressed buffers are more than 2 GB in size, things are going to break in general due to CLR limitations. That's a much larger topic.